src/rust/engine/logging/src/lib.rs

```diff
@@ -46,8 +46,38 @@ macro_rules! debug_log {
     };
 }
 
+use enumflags2::BitFlags;
+use enumflags2_derive::EnumFlags;
+use futures::task_local;
+use parking_lot::Mutex;
+
+// These values are hard-coded on the Python side in native's override_thread_logging_destination_*
+// methods. Ensure any updates here are reflected there.
+#[derive(EnumFlags, Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(u8)]
+pub enum Destination {
+  Pantsd = 0b01,
+  Stderr = 0b10,
+}
+
+thread_local! {
+  pub static THREAD_DESTINATION: Mutex<BitFlags<Destination>> = Mutex::new(Destination::Stderr.into())
+}
+
+task_local! {
+  static TASK_DESTINATION: Mutex<Option<BitFlags<Destination>>> = Mutex::new(None)
+}
+
+pub fn set_task_logging_destination(destination: BitFlags<Destination>) {
```

Comment:
A doc blob somewhere that explains the overall approach (starting by inlining the PR description) would be handy.

Reply:
Done (on the Destination enum)